### PR TITLE
chore: enable prettier

### PR DESCRIPTION
**Pull request by kevgo**
As agreed in https://github.com/ory/docs/pull/862, this enables Prettier for the entire repo and formats all the things in it. I am enabling the override to 130 characters text width for Markdown files to conform to @tomekpapiernik's current style and get this merged with the least amount of changes. We can adjust details later once Prettier is enabled.

I have reviewed all changed files manually, all of them look safe to me.

Some comments:

- Prettier has found a number of bugs in Markdown, like unclosed emphasis or missing indentation of text blocks.
- I have adjusted the indentation of all list elements that contain embedded code blocks. @tomekpapiernik please double-check that the indentation of files that you worked on is still okay.
- Changes to YML and JS files are extremely safe, no need to review them.
- We are changing single-quotes to double-quotes in JS and TS to make these languages consistent with the other languages we use (Go, HTML, JSON) which all use double-quotes.
- all tests are still passing  ✅ 
- the vast majority of these changes improves readability and consistency in my opinion

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: docs/keto/sdk/go.md
Comment (kevgo) [inline docs/keto/sdk/go.md:6]: If we rename this file to `go.mdx`, then Prettier doesn't merge these two lines. I think given that this file contains Gatsby stuff it should have that extension anyways, no?

Comment (aeneasr) [reply docs/keto/sdk/go.md:6]: 👍

**Review by tomekpapiernik — CHANGES_REQUESTED** [review]
Hey @kevgo, the PR looks pretty good from my point of view. The group of documents I consider the most problematic (social sign-in) looks good aside from the Facebook document. Could you please fix that one? As you can see in the screenshot, the list styling is broken for some reason. 

I would also appreciate it if you could let me know what kind of indentations I should use to get code block / sub-list -> list item affinity you can see across the social sign-in docs. I think this kind of formatting will come in handy from time to time and I'd like to avoid fighting prettier when I decide to use it. 

Additionally, I'm not so sure about formatting the autogenerated CLI documents. These are generated from the CLI source code and committed to the docs repo by a bot. As such, they will be pushed to the repo in the unformatted state and will overwrite the previously formatted versions. Formatting will have to be triggered manually to fix that. The formatted state won't persist for long - only until the next automated CLI reference regen. 

This leaves me wondering - wouldn't it be better to either leave the CLI reference documentation out of the formatter's scope or to try and keep the formatting in check on the CLI code level?

File: docs/keto/sdk/go.md
Comment (tomekpapiernik) [reply docs/keto/sdk/go.md:6]: Maybe this PR is a good opportunity to make sure all files use .mdx extensions instead of .md? 🤔

Comment (kevgo) [reply docs/keto/sdk/go.md:6]: That's a separate change from enabling Prettier, so I think it deserves a separate PR.

And given how massive this PR is, I would prefer to get it merged with the fewest changes possible. What I want to avoid is that anybody has to review 329 files again. 

I'm happy to submit a new PR that fixes file extensions right after this one goes in.

File: docs/guides/_common/oidc_session_hook.mdx
Comment (kevgo) [reply docs/guides/_common/oidc_session_hook.mdx:11]: It's the some content, just formatted differently ([proof](https://yaml-online-parser.appspot.com/?yaml=parent_a%3A%0A-+one%0A-+two%0Aparent_b%3A%0A++-+one%0A++-+two&type=json))

File: docs/guides/social-signin/facebook.mdx
Comment (kevgo) [reply docs/guides/social-signin/facebook.mdx]: Good catch! I have fixed this in https://github.com/ory/docs/pull/888/commits/b2c9ac5b17091c6b645690384a6bd8b0c59573e2. Please check again.

**Comment by kevgo** [comment]
Thanks for the review and the helpful comments @tomekpapiernik! In general, the indentation for content inside lists should be so deep that it aligns with the beginning of the text. This [example](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBJKACA5guAnAQwBsAaDABTzhhgEt8NaoATBGAZw0ljcandqsMRWuw4Z2EDDAAWBGBmJFpMuNLgAPBaIwA3UQFclAT0UisUOMwB0GABL41BKoowAzOAHcMmggFsAByI4diQAHSgIiIBaDAAVVXdaPDF1LS45QjB4PAwINwwDKAg8VipmYVFxAAoCAGsCSrFOT1pZDAAjAyJghQCIJg4ASkZOACYMfoE6aE4CTwJTNzwIPxU1DrgsJigmLDyC2TURSzJJdbTtTiZWHgrW9phPKXYAgjAQxRa4HusozHiMh0OhgeAMajcJQkBg67DgAEdwbBJs4CFhCAEZJxZPJOj9oPsYFIjk1tPA-LYEnBTM4nERzjc2FYMA8ZNJnhI3h92LYIhgMBEAIyUxJuZKpeDpMCZd45A55UqOCoiZoYWoNDBQAx+TblUnsEY6ADMkwg01os0UCyWKzWJM22yguyg+3yFxOcBFaklV3WVEhLkZdxZbTZsioale7xCf0iAMBwOxYIhUPYMLhiN4b0I6IImOxcgUmyIBOkxKBnBVZLgFMB1MULmIDJYTPuoZUEc50Z5fP5EQcgbmmE0oJral8gWC4XjQoADLOAJzz5cr1fz2yoQ6JKuMcmjDBUEzCZw4Mgk7NojFs4Et4METh+AzSr77yxWZlEjB+erexI+zolmAdRmLQFiTCsAT4EQpgAFYGKkIh1GobQsvefAXAEVD6BA8EojmV49nG-LESRpFkfyCSJtIybuKm6YIkifSorm+Yvp4PxEBumCyDoYD3nAZDsRgci6N6UjEKB3GqGsrLsmwphRtyACEEQgCQIAQAEMz8MgoDOCsnjkLSOkoMQ1rsGpIAdFkSEwAAyv4cAADJMHAyBuE2AlWTZ1B2VyezIKC4LqTWmzMKwzBOQQLpGDgABiJTfjQAWmQYRKWTIMB+EQADqQLwIpcB2Qg5r6DAxjIOA7AWepTBwngMCUGi37uZ56kwewGh2XswQAIoGBA8CtfSXnZvVlXfngdTMBAnhQJZmGDDlgiyMgAAcs7qZhZpwDlGKVWZiw1SAiKDXAlCaRZpnsNEb4RZZVCIsk505i1SAeSN6lwn4tCBcmX09XA-VncNcLqTABAdMtzCrUgYzg4QtAnFgADCqxvSAIQAKyWfBcBxJDV0fWDIC6OC6C3PZYB4LQWkAIIsHZ5XBKDcAAL5s0AA) hopefully makes this more clear and visual.

When I am not sure, I indent content too much (4 spaces) and let Prettier auto-corrects the indentation to the "correct" width.

**Comment by kevgo** [comment]
All comments addressed, please take another look.

File: docs/keto/sdk/go.md
Comment (tomekpapiernik) [reply docs/keto/sdk/go.md:6]: Fair point. This is something either of us can handle when this one is merged.

**Comment by tomekpapiernik** [comment]
@kevgo thanks for the update. Unfortunately, one thing got fixed and another one broke 😅 The line of text highlighted in the screenshot is supposed to be an imported admonition. From my experience, these imports break whenever you try to "attach" them to list items and it seems that there's no way around it. As such, these have to remain unaffiliated to be properly displayed. 

image


Also, there's a point you didn't comment on. Let me quote myself: 

> Additionally, I'm not so sure about formatting the autogenerated CLI documents. These are generated from the CLI source code and committed to the docs repo by a bot. As such, they will be pushed to the repo in the unformatted state and will overwrite the previously formatted versions. Formatting will have to be triggered manually to fix that. The formatted state won't persist for long - only until the next automated CLI reference regen.

**Comment by kevgo** [comment]
Thanks for the detailed bug report and the guidance on how to fix it @tomekpapiernik. I have scanned all files for this issue and this seems to be the only one where I indented such import statements. After the fix, it seems to look okay on my machine.

image

Sorry about forgetting to comment on formatting auto-generated content. I would not format it using Prettier. Reasons are:

- Prettier is very slow, let's not make it slower by have it format files that it doesn't need to.
- Prettier formatting is to help humans feel at home in each other's code. Auto-generated code is usually not looked at or edited by humans.

We can tell Prettier to ignore certain files or folders by adding them to the [.prettierignore](https://github.com/kevgo-ory/docs/blob/kg-enable-prettier/.prettierignore) file. I have already added a few entries to it. Feel free to add entries for the auto-generated files after this PR is merged!

**Comment by kevgo** [comment]
The fix for the indentation is at https://github.com/ory/docs/pull/888/commits/4369eefb526a9cc00c40f337f2d808e5c4fb2c0a

**Review by tomekpapiernik — DISMISSED** [review]
Thanks for the explanations @kevgo. Everything looks good to me right now. Please update the branch (I can't do it for some reason - did you allow maintainers to push to your branch?) and merge it to master :) If you can't merge then just update and let me know, I'll merge that 4 u. Cheers!

**Review by tomekpapiernik — APPROVED** [review]

**Comment by kevgo** [comment]
Nice collaboration on this @tomekpapiernik. At the risk of repeating myself, Prettier is most useful when you install its IDE plugins and configure them to format everything each time you save a file.

- VSCode: https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode
- IntelliJ (I think, haven't used it): https://www.jetbrains.com/help/idea/prettier.html#ws_prettier_reformat_code

**Comment by tomekpapiernik** [comment]
Thanks, I'll have a look at the VSC extension but I have no problems running the formatting job manually :) 
I opened a follow-up to this PR, please have a look @kevgo 
https://github.com/ory/docs/pull/896

**Pull request merged**
